go/cli/mcap/utils/http_seeker.go

```diff
@@ -0,0 +1,237 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strconv"
+	"syscall"
+)
+
+// Interval represents a range of downloaded bytes.
+type Interval struct {
+	Start int64 // Inclusive
+	End   int64 // Exclusive
+}
+
+type Option func(*HTTPSeeker)
+
+func WithMinRequestSize(size int64) Option {
+	return func(hs *HTTPSeeker) {
+		hs.minRequestSize = size
+	}
+}
+
+func WithHeaders(headers http.Header) Option {
+	return func(hs *HTTPSeeker) {
+		hs.headers = headers
+	}
+}
+
+type HTTPSeeker struct {
+	url            string
+	size           int64
+	pos            int64
+	minRequestSize int64
+	headers        http.Header
+	buffer         []byte
+	intervals      []Interval
+}
+
+func NewHTTPSeeker(url string, opts ...Option) (*HTTPSeeker, error) {
+	hs := &HTTPSeeker{
+		url:            url,
+		minRequestSize: 32 * 1024, // Default 32KB minimum request size
+		headers:        make(http.Header),
+		intervals:      []Interval{},
+	}
+
+	for _, opt := range opts {
+		opt(hs)
+	}
+
+	// Get file size
+	req, err := http.NewRequest("HEAD", url, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range hs.headers {
+		req.Header[k] = v
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	hs.size, err = strconv.ParseInt(resp.Header.Get("Content-Length"), 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create memory mapping
+	hs.buffer, err = syscall.Mmap(
```

Comment:
I get the appeal of using a big mapped address space region to store the downloaded MCAP data, but there are some issues with this approach:
1. We're now using the syscall module directly, which as the module is documented means we need to test changes on every platform
1. Say I'm running `mcap cat` on a 10GB remote file, I want to be dumping my processed result to stdout without keeping it all in memory as well. Since `cat` reads the file from front to back, it will dirty every page of this buffer.

Reply:
>  It makes a series of 8 byte range requests as mcap.Reader reads the magic, then header etc. which is wasteful in terms of round trips so I am going to re-implement that with a minimum HTTP request size. Should I add that to this PR or a follow up?

I think leave it out of this PR.